### PR TITLE
Updating JSON.Net & FluentSimulator. Using newer MS Docker repo

### PR DESCRIPTION
**Pull request by matt-lethargic**
## Context

Updating Json.Net to v12 out of v12-beta
Updating FluentSimulator to v2 as v1 did not support .Net Standard
Changing Dockerfile to use Microsofts repo

## Changes proposed in this pull request

Keeps the system up-to-date, stops compatibility issues with old packages

### CODE REVIEW

File: CryptoTechReminderSystem.Test/Gateway/HarvestGatewayTests.cs [hunk 1]
```diff
@@ -15,13 +15,48 @@ public class HarvestGatewayTests
...
             @"Software Engineer, Senior Software Engineer, Senior Engineer, Lead Engineer, 
             Delivery Manager, SRE, Consultant, Delivery Principal";
-        
+
         private static FluentSimulator _harvestApi;
         private static HarvestGateway _harvestGateway;
-        
+
+        [TestFixture]
+        public class CanContrsuctInstance
```
Comment (yusufsheiqh) [inline CryptoTechReminderSystem.Test/Gateway/HarvestGatewayTests.cs:26]:
```suggestion
        public class CanConstructInstance
```

File: CryptoTechReminderSystem.Test/Gateway/HarvestGatewayTests.cs [hunk 2]
```diff
@@ -15,13 +15,48 @@ public class HarvestGatewayTests
...
+        {
+            [Test]
+            public void CanUseNullRoles()
+            {
+                string roles = null;
+
+                HarvestGateway gateway = new HarvestGateway(Address, Token, HarvestAccountId, UserAgent, roles);
+
+                Assert.IsNotNull(gateway);
```
Comment (yusufsheiqh) [inline CryptoTechReminderSystem.Test/Gateway/HarvestGatewayTests.cs:35]:
```suggestion
                gateway.Should().NotBeNull();
```

File: CryptoTechReminderSystem.Test/Gateway/HarvestGatewayTests.cs [hunk 3]
```diff
@@ -15,13 +15,48 @@ public class HarvestGatewayTests
...
+            }
+
+            [Test]
+            public void ThrowsExceptionWhenAddressNull()
+            {
+                var exception = Assert.Throws<ArgumentNullException>(()=> new HarvestGateway(null, Token, HarvestAccountId, UserAgent, BillablePersonRoles));
+                Assert.AreEqual("address", exception.ParamName);
```
Comment (yusufsheiqh) [inline CryptoTechReminderSystem.Test/Gateway/HarvestGatewayTests.cs:42]:
```suggestion
                exception.ParamName.Should().Be("address");
```

File: CryptoTechReminderSystem.Test/Gateway/HarvestGatewayTests.cs [hunk 4]
```diff
@@ -15,13 +15,48 @@ public class HarvestGatewayTests
...
+            }
+
+            [Test]
+            public void ThrowsExceptionWhenTokenNull()
+            {
+                var exception = Assert.Throws<ArgumentNullException>(()=> new HarvestGateway(Address, null, HarvestAccountId, UserAgent, BillablePersonRoles));
+                Assert.AreEqual("token", exception.ParamName);
```
Comment (yusufsheiqh) [inline CryptoTechReminderSystem.Test/Gateway/HarvestGatewayTests.cs:49]:
```suggestion
                exception.ParamName.Should().Be("token");
```

File: CryptoTechReminderSystem.Test/Gateway/HarvestGatewayTests.cs [hunk 5]
```diff
@@ -15,13 +15,48 @@ public class HarvestGatewayTests
...
+            }
+
+            [Test]
+            public void ThrowsExceptionWhenAccountIdNull()
+            {
+                var exception = Assert.Throws<ArgumentNullException>(() => new HarvestGateway(Address, Token, null, UserAgent, BillablePersonRoles));
+                Assert.AreEqual("accountId", exception.ParamName);
```
Comment (yusufsheiqh) [inline CryptoTechReminderSystem.Test/Gateway/HarvestGatewayTests.cs:56]:
```suggestion
                exception.ParamName.Should().Be("accountId");
```

**Review by yusufsheiqh — COMMENTED** [review]
A few minor changes where we are opting to use Fluent Assertions, but otherwise looks great to me, and @chubberlisk!

**Review by chubberlisk — APPROVED** [review]
:shipit:

(@yusufsheiqh and I have had a reviewaroo and looks good to us. 🎉  Thank you for the PR!)

**Review by yusufsheiqh — APPROVED** [review]

**Pull request merged**